### PR TITLE
Fugitives - Mid Round Antag

### PR DESCRIPTION
**Pull request by PHCodes**
## About the PR

Added Fugitives and support for the Mid Round Antags system. 

## Why / Balance

We didn't have Fugitives yet. 

## Technical details

I had to make several changes along the way, some of the largest notes being:
Job is now JobComponent and just takes a string as input. 
TryGetMind outputs the exact mind component we care about. I'm not sure the enclosing query loop even does anything anymore. 
Objectives are now a component that you fetch. This was the biggest change from Nyanotrasen's code. You fetch the Objective and then read out the information that's relevant for marking the entity as an antagonist with the role it should have, etc. I imagine a similar change will need to be made for all mid round antags. 
## Media

image
https://imgur.com/a/69b1uXX
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: JJ
- add: Added Fugitives and Midround Antag system.

### CODE REVIEW

**Review by DebugOk — CHANGES_REQUESTED** [review]
This should be drafted until metempsychosis is in, instead of adding a bunch of unrelated stuff

**Comment by deltanedas** [comment]
why doesnt this use generic antag + objectives round end text, like ninja

File: Content.IntegrationTests/Tests/Nyanotrasen/NyanoRandomTests.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                        Assert.IsTrue(prototypeManager.HasIndex<SpeciesPrototype>(proto),
+                            $"{proto} is in MetempsychoticHumanoidPool but is not defined anywhere.");
+                    }
+                }
+            });
+
+            await pairTracker.CleanReturnAsync();
+        }
+    }
+}
```
Comment (DEATHB4DEFEAT) [inline Content.IntegrationTests/Tests/Nyanotrasen/NyanoRandomTests.cs:119]:
```suggestion
}

```

File: Resources/Prototypes/Nyanotrasen/Entities/Mobs/Player/humanoid.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  parent: MobHuman
+  id: MobHumanFugitive
+  name: Fugitive
+  noSpawn: true
+  components:
+    - type: Loadout
+      prototypes: [FugitiveStartingGear]
+    - type: Fugitive
+    - type: FugitiveCountdown
+    - type: RandomHumanoidAppearance
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/Nyanotrasen/Entities/Mobs/Player/humanoid.yml:11]:
```suggestion
    - type: RandomHumanoidAppearance

```

File: Resources/Textures/Nyanotrasen/Structures/Machines/metempsychotic.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      "delays": [
+        [
+          0.1,
+          0.1,
+          0.1,
+          0.1,
+          0.1,
+          0.1
+        ]
+      ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/Nyanotrasen/Structures/Machines/metempsychotic.rsi/meta.json:24]:
```suggestion
      "delays": [[ 0.1, 0.1, 0.1, 0.1, 0.1, 0.1 ]]
```

File: Resources/Prototypes/Nyanotrasen/metempsychoticHumanoids.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+- type: weightedRandom
+  id: MetempsychoticHumanoidPool # Species prototypes
+  weights:
+    Human: 1
+    Felinid: 1
+    Oni: 1
+    Arachnid: 1
+    Moth: 1
+    #Harpy: 1 #Still need to be merged.
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/Nyanotrasen/metempsychoticHumanoids.yml:9]:
```suggestion
    Harpy: 1
```

File: Resources/Prototypes/Nyanotrasen/metempsychoticHumanoids.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  id: MetempsychoticHumanoidPool # Species prototypes
+  weights:
+    Human: 1
+    Felinid: 1
+    Oni: 1
+    Arachnid: 1
+    Moth: 1
+    #Harpy: 1 #Still need to be merged. 
+    Reptilian: 0.5
+    #Dwarf: 0.5 #'Dwarves are a no no'
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/Nyanotrasen/metempsychoticHumanoids.yml:11]:
```suggestion
    # Dwarf: 0.5 # "Dwarves are a no no"
```

File: Resources/Locale/en-US/nyanotrasen/station-events/events/fugitive.ftl [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+fugitive-user-was-a-fugitive = [color=gray]{$user}[/color] was a fugitive.
+fugitive-user-was-a-fugitive-named = [color=White]{$name}[/color] ([color=gray]{$user}[/color]) was a fugitive.
+fugitive-was-a-fugitive-named = [color=White]{$name}[/color] was a fugitive.
+
+fugitive-user-was-a-fugitive-with-objectives = [color=gray]{$user}[/color] was a fugitive who had the following objectives:
+fugitive-user-was-a-fugitive-with-objectives-named = [color=White]{$name}[/color] ([color=gray]{$user}[/color]) was a fugitive who had the following objectives:
+fugitive-was-a-fugitive-with-objectives-named = [color=White]{$name}[/color] was a fugitive who had the following objectives:
+JobFugitive = Fugitive
```
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/nyanotrasen/station-events/events/fugitive.ftl:31]:
```suggestion
JobFugitive = Fugitive

```

File: Resources/Locale/en-US/nyanotrasen/medical/components/metempsychotic.ftl [hunk 7]
```diff
@@ -0,0 +1 @@
+medical-scanner-upgrade-metem = Metempsychotic machine karma bonus
```
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/nyanotrasen/medical/components/metempsychotic.ftl:1]:
```suggestion
medical-scanner-upgrade-metem = Metempsychotic machine karma bonus

```

File: Resources/Prototypes/Nyanotrasen/Objectives/fugitiveObjective.yml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+- type: entity
+  id: EscapeShuttleObjectiveFugitive
+  name: Escape, fugitive, to Centcomm alive and unrestrained.
+  description: You need to escape the station, one way or another. 
+  components:
+  - type: Objective
+    difficulty: 69 #'Make this number absurdly high' - 10/17/2023
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/Nyanotrasen/Objectives/fugitiveObjective.yml:7]: I don't think dates know how to speak
```suggestion
    difficulty: 69 # "Make this number absurdly high" - 10/17/2023
```

File: Content.Shared/Mind/SharedMindSystem.cs [hunk 9]
```diff
@@ -38,6 +39,16 @@ public override void Initialize()
...
+    [PublicAPI]
+    //Nyano - Code Block Summary: Allows for us to set if a mind can be examined or not. Used for Metem cloning. 
+    public void SetExamineInfo(EntityUid uid, bool canExamine, MindContainerComponent? mind = null)
+    {
+        if (!Resolve(uid, ref mind, false))
+            return;
+
+        mind.ShowExamineInfo = canExamine;
+    }
+    //Nyano - End Code Block.
```
Comment (DEATHB4DEFEAT) [inline Content.Shared/Mind/SharedMindSystem.cs:51]:
```suggestion
    //Nyano - End Code Block.

```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Nyanotrasen/Objectives/fugitiveObjective.yml (hunk 8 above)
Comment (deltanedas) [reply Resources/Prototypes/Nyanotrasen/Objectives/fugitiveObjective.yml:7]: also it wouldnt matter since difficulty is only for tator

**Pull request closed without merging**